authz/providers/azure/azure.go

```diff
@@ -120,6 +124,8 @@ func (s Authorizer) Check(ctx context.Context, request *authzv1.SubjectAccessRev
 		return &authzv1.SubjectAccessReviewStatus{Allowed: true, Reason: rbac.AccessAllowedVerdict}, nil
 	}
 
+	ctx = azureutils.WithRetryableHttpClient(ctx, s.HttpClientRetryCount)
```

Comment:
1. why do we save the http client in the context?
2. as far as I can tell, it still uses [AccessInfo.client](https://github.com/kubeguard/guard/blob/master/authz/providers/azure/rbac/rbac.go#L160), no?

Reply:
I followed the comment of `WithRetryableHttpClient`.   @AzureMarker **Some of the libraries** respects the http client in the context. Does it mean that it's a best-effort?

```
// WithRetryableHttpClient sets the oauth2.HTTPClient key of the context to an
// *http.Client made from makeRetryableHttpClient.
// Some of the libraries we use will take the client out of the context via
// oauth2.HTTPClient and use it, so this way we can add retries to external code.
```

Reply:
`net/http` is a standard library, it doesn't reference third party code so there's no way it uses that context key.

This key is used in go-oidc:
https://github.com/kubeguard/guard/blob/e0840f9ce43103cd006c07b1cdb4373125956d3e/vendor/github.com/coreos/go-oidc/oidc.go#L60

The other HTTP client usages hard-code to `go.kubeguard.dev/guard/util/httpclient`'s `DefaultHTTPClient`, which is just a normal HTTP client.

Reply:
I made this change https://github.com/kubeguard/guard/pull/372/commits/57e588e83569f709f8fd04b22b8dadca66c24080. Could you please review it?

Reply:
@AzureMarker gentle ping

Reply:
Added some comments